core/src/main/java/org/pentaho/di/connections/vfs/BaseVFSConnectionProvider.java

```diff
@@ -90,4 +101,65 @@ protected static boolean getBooleanValueOfVariable( VariableSpace space, String
   protected VariableSpace getSpace( ConnectionDetails connectionDetails ) {
     return connectionDetails.getSpace() == null ? Variables.getADefaultVariableSpace() : connectionDetails.getSpace();
   }
+
+  @Override
+  public boolean test( T connectionDetails, VFSConnectionTestOptions connectionTestOptions ) throws KettleException {
+    boolean valid = test( connectionDetails );
+    if ( !valid ) {
+      return false;
+    }
+
+    if ( !connectionDetails.isSupportsRootPath()  || connectionTestOptions.isIgnoreRootPath() ) {
+      return true;
+    }
+
+    String resolvedRootPath = getResolvedRootPath( connectionDetails );
+    if ( StringUtils.isEmpty( resolvedRootPath ) ) {
+      return !connectionDetails.isRootPathRequired();
+    }
+
+    String internalUrl = buildUrl( connectionDetails, resolvedRootPath );
+    FileObject fileObject = KettleVFS.getFileObject( internalUrl, new Variables(), getOpts( connectionDetails ) );
+
+    try {
+      return fileObject.exists();
+    } catch ( FileSystemException fileSystemException ) {
+      LOGGER.error( fileSystemException.getMessage() );
+      return false;
+    }
+  }
```

Comment:
Another point of discussion, How/where is `#test(ConnectionDetails, VFSConnectionTestOptions)` being called?

Currently PDI File Open Browser has test functionality and it's calling

`result = connectionManager.test( connectionDetails );`
https://github.com/pentaho/pentaho-kettle/blob/7bb5b14e652642ed613e49db3fdfece1186088c1/plugins/connections/ui/src/main/java/org/pentaho/di/vfs/connections/ui/dialog/ConnectionDialog.java#L408-L414

Then the code does to ConnectionManager#test

https://github.com/pentaho/pentaho-kettle/blob/7bb5b14e652642ed613e49db3fdfece1186088c1/core/src/main/java/org/pentaho/di/connections/ConnectionManager.java#L288-L292


I wouldn't want to have to cast down to BaseVFSConnectionProvider to execute the rootPath test. 

Shouldn't this be more of the approach

```
@Override
public boolean test( T ConnectionDetails) {
  VFSConnectionTestOptions testOptions =  // create options based on ConnectionDetails
  test(connectionDetails,  testOptions)
}
```